components/AppLayout/GlobalSearch.js

```diff
@@ -60,6 +66,14 @@ const useStyles = makeStyles(theme => ({
     display: 'flex',
     alignItems: 'center',
   },
+  ellipsisOnMobile: {
+    [theme.breakpoints.down('md')]: {
+      maxWidth: 120,
```

Comment:
1. Seems that ellipsis is also needed on desktop 
![image](https://user-images.githubusercontent.com/108608/85193429-0a71ff80-b2fb-11ea-9480-ed215110ee79.png)
2. In order to trigger overflow, we should set `min-width: 0` to override the box's [default min-sizing behavior](https://css-tricks.com/flexbox-truncated-text/) (which enforces the flex children to have same width as content, thus no overflow could happen). In this case, we won't need to set a fixed `max-width`, and the style should work on different screen widths.
![image](https://user-images.githubusercontent.com/108608/85193465-5c1a8a00-b2fb-11ea-8ab1-4f8ff86c5167.png)